### PR TITLE
Add Octavia HM port to network configs

### PR DESCRIPTION
**Pull request by hemanthnakkina**
This patch adds Octavia Health manager IP address to the
network configs section if o-hm0 port exists. If there is
no IP address to o-hm0, will be notified in potential issues.

Resolves: #122

### CODE REVIEW

File: plugins/openstack/pyparts/network.py [hunk 1]
```diff
@@ -69,6 +73,17 @@ def get_config_network_info(self):
             config_info['nova'][key] = "{} ({})".format(port.addresses[0],
                                                         port.name)
 
+        for key, port in self.octavia_bind_interfaces.items():
+            if 'octavia' not in config_info:
+                config_info['octavia'] = {}
+
+            if len(port.addresses) > 0:
```
Comment (dosaboy) [inline plugins/openstack/pyparts/network.py:80]: you can shorten this to just 'if port.addresses:'

File: plugins/openstack/pyparts/network.py [hunk 2]
```diff
@@ -69,6 +73,17 @@ def get_config_network_info(self):
...
+                config_info['octavia'][key] = "{} ({})".format(
+                        port.addresses[0], port.name)
```
Comment (dosaboy) [inline plugins/openstack/pyparts/network.py:82]: This assumes that the interface will only ever have one address which afaik is fine.

File: plugins/openstack/pyparts/network.py [hunk 3]
```diff
@@ -69,6 +73,17 @@ def get_config_network_info(self):
...
+            else:
+                msg = ("Octavia Health manager port has no IP address")
```
Comment (dosaboy) [inline plugins/openstack/pyparts/network.py:84]: Lets elaborate this a bit e.g.:

"No IP address found on Octavia healthmanager port ({}). Octavia will not be able to communicate with Amphora vms - please investigate.".format(port.name)

**Review by dosaboy — CHANGES_REQUESTED** [review]

File: core/plugins/openstack/__init__.py [hunk 4]
```diff
@@ -287,6 +287,19 @@ def neutron_bind_interfaces(self):
...
+    @property
+    def octavia_bind_interfaces(self):
+        """
+        Fetch interface o-hm0 used by Openstack Octavia. Returned dict is
+        keyed by config key used to identify interface.
+        """
+        interfaces = {}
+        port = self.nethelp.get_interface_with_name('o-hm0')
+        if port:
+            interfaces.update({'o-hm0': port})
```
Comment (dosaboy) [inline core/plugins/openstack/__init__.py:299]: Is there are an actual config key name we can use instead of "o-hm0"?

**Review by dosaboy — CHANGES_REQUESTED** [review]

File: plugins/openstack/pyparts/network.py (hunk 1 above)
Comment (hemanthnakkina) [reply plugins/openstack/pyparts/network.py]: Done

File: plugins/openstack/pyparts/network.py (hunk 2 above)
Comment (hemanthnakkina) [reply plugins/openstack/pyparts/network.py:82]: Yes

File: plugins/openstack/pyparts/network.py (hunk 3 above)
Comment (hemanthnakkina) [reply plugins/openstack/pyparts/network.py]: o-hm0 is hardcoded in charm-octavia https://opendev.org/openstack/charm-octavia/src/branch/master/src/lib/charm/openstack/octavia.py#L39

File: core/plugins/openstack/__init__.py (hunk 4 above)
Comment (hemanthnakkina) [reply core/plugins/openstack/__init__.py:299]: o-hm0 is hardcoded in charm-octavia https://opendev.org/openstack/charm-octavia/src/branch/master/src/lib/charm/openstack/octavia.py#L39

**Review by dosaboy — APPROVED** [review]
lgtm thanks hemanth!
